packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationRegistry.m

```diff
@@ -79,6 +90,11 @@ - (void)setMask:(UIInterfaceOrientationMask)mask forModule:(id)module {
 
 - (UIInterfaceOrientationMask)requiredOrientationMask
 {
+  // The app is moved to the foreground.
+  if (!_foregroundedModule && _lastOrientationMask) {
```

Comment:
If there is no `_foregroundedModule`, but `_lastOrientationMask` is falsy, we will skip this clause, move over to `[_moduleInterfaceMasks objectForKey:_foregroundedModule]` which will result in `nil`, since `_foregorundedModule` is `nil` and return `0`.

Was this `&& _lastOrientationMask` intentional?

Reply:
So, I wanted to make it less confusing by adding this check. When `_lastOrientationMast` is `0` and the `_foregroundedModule` is `nil` we still returns `0` as a default value. But not as a saved `_lastOrientationMask`. I'm not sure if you will understand what I mean by this. However, you pointed out this so it is still confusing 😅

Reply:
No, actually I understood the code as you wanted to write it, just wanted to make sure. Good job! (The `if` handles only one, particular situation, all the other situations are handled below.)